### PR TITLE
VZ-9835: Activate kontainer object for ociocne driver when capi is enabled

### PR DESCRIPTION
**Pull request by mgianatagh**
Provide a summary of the change and which issue (i.e. ticket) is fixed.
If there are any dependencies, for example on a PR in another repository, list those.

### CODE REVIEW

File: platform-operator/verrazzano-bom.json [hunk 1]
```diff
@@ -151,13 +151,13 @@
               "rancherUI": "v2.7.2-2/release-2.7.2",
               "ocneDriverVersion": "v0.2.0",
               "ocneDriverChecksum": "60cf8205a681a53ab73e6e9d125e741afa681b54cfd72d6c14e906d3e8b71f0c",
-              "tag": "v2.7.3-20230526171959-8e91f98f0",
+              "tag": "20230527191156-86dc4b493",
```
[the remark this answers is not in the document]
Comment (mgianatagh) [reply platform-operator/verrazzano-bom.json:154]: Yes, I will first need to rebuild Rancher once those changes are merged.

File: platform-operator/controllers/verrazzano/component/capi/capi.go [hunk 2]
```diff
@@ -249,3 +270,87 @@ func applyTemplate(templateContent string, params interface{}) (bytes.Buffer, er
 	// Return the result containing the processed template
 	return buf, nil
 }
+
+// createOrUpdateKontainerCR - Create or update the kontainerdrivers.management.cattle.io object that
+// registers the ociocne driver
+func createOrUpdateKontainerCR(ctx spi.ComponentContext) error {
```
[the remark this answers is not in the document]
Comment (mgianatagh) [reply platform-operator/controllers/verrazzano/component/capi/capi.go:276]: Interesting, I thought it was being created by this method when I was doing my testing.

File: platform-operator/verrazzano-bom.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmaron99) [reply platform-operator/verrazzano-bom.json:154]: OK - well I approved the Rancher changes so will approve this one once the build is done there.  I assume @michael-cico has seen the bom related changes prior (new ocne driver related entries)?

File: platform-operator/controllers/verrazzano/component/capi/capi.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jmaron99) [reply platform-operator/controllers/verrazzano/component/capi/capi.go:276]: Hmm..in looking at the code I didn't see an update.  Maybe I didn't look close enough..

[the remark this answers is not in the document]
Comment (jmaron99) [reply platform-operator/controllers/verrazzano/component/capi/capi.go:276]: I guess technically the driver resource may get updated, and it is referenced by the kontainer

**Review by anders-swanson — DISMISSED** [review]

**Review by anders-swanson — DISMISSED** [review]

File: tests/e2e/verify-install/capi/capi_test.go [hunk 3]
```diff
@@ -68,10 +76,115 @@ func WhenCapiInstalledIt(description string, f func()) {
 	})
 }
 
+var _ = t.Describe("KontainerDriver status", Label("f:platform-lcm.install"), func() {
```
Comment (jmaron99) [inline tests/e2e/verify-install/capi/capi_test.go:79]: Should we add a post-upgrade test somewhere or are we comfortable it's essentially the same code path?

**Review by jmaron99 — APPROVED** [review]

**Pull request merged**